### PR TITLE
Changes to System.Xml.XmlSerializer

### PR DESCRIPTION
**Pull request by venkat-raman251**
@weshaggard ,@mellinoe : Please have a look.

### CODE REVIEW

File: src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj [hunk 1]
```diff
@@ -11,9 +11,12 @@
     <DebugSymbols>true</DebugSymbols>
     <PlatformTarget>AnyCPU</PlatformTarget>
     <ProjectGuid>{D62A6082-5229-4845-8BE9-75753E08C65A}</ProjectGuid>
+    <DefineConstants Condition="'$(TargetGroup)' == 'netcore50'">$(DefineConstants);NET_NATIVE</DefineConstants>
```
Comment (mellinoe) [inline src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj]: I see that the internal build also has `<EnableWinRT>true</EnableWinRT>` in the project file. Do we need that here?

**Comment by mellinoe** [comment]
There is also an rd.xml file for .NET Native in the internal build, which is included as an EmbeddedResource for .NET Native.

````XML
<ItemGroup>
  <EmbeddedResource Include="$(MsBuildThisFileDirectory)Resources\$(AssemblyName).rd.xml" />
</ItemGroup>
````

File: src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj (hunk 1 above)
Comment (venkat-raman251) [inline src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj]: Wes asked me to remove it as it wasnt used anywhere in the code.

Comment (mellinoe) [inline src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj]: Sounds good to me.

File: src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj [hunk 2]
```diff
@@ -76,6 +79,9 @@
     <Compile Include="System\Xml\Serialization\SourceInfo.cs" />
     <Compile Include="System\Xml\Serialization\XmlSchema.cs" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)' == 'netcore50'">
```
Comment (mellinoe) [inline src/System.Xml.XmlSerializer/src/System.Xml.XmlSerializer.csproj]: nit: 2 spaces of extra indentation

**Comment by mellinoe** [comment]
LGTM

**Comment by venkat-raman251** [comment]
@dotnet-bot Test Innerloop Windows_NT Debug Build and Test

**Comment by venkat-raman251** [comment]
@weshaggard : Can this be merged?

**Comment by mellinoe** [comment]
LGTM

**Comment by weshaggard** [comment]
LGTM.

cc @khdang @SGuyGe as FYI

**Pull request merged**